Zend/zend_operators.h

```diff
@@ -138,9 +138,40 @@ static zend_always_inline bool zend_is_long_compatible(double d, zend_long l) {
 	return (double)l == d;
 }
 
+static zend_always_inline bool zend_scalar_is_valid_bool(zval *arg) {
+	bool result = 0;
+
+	switch (Z_TYPE_P(arg)) {
+		case IS_TRUE:
+		case IS_FALSE:
+			result = 1;
+			break;
```

Comment:
```suggestion
			return true;
```

Reply:
I mimicked `i_zend_is_true` and find that only returning once at the end seems clearer - or what are the benefits to this?

Reply:
compiler will probably optimize the variable assigment away, but if we can return /wo extra variable assigment, I think it is better and slighly more readable, let's listen to the feedback from others :)

Reply:
Removing the temporary variable and only using returns seems like a sensible improvement, I will do that, thanks!